### ISSUE TITLE
Add Tags to the Storefront filtering

### ISSUE THREAD

**Issue by appscompany** [Shopify/dawn#333]
**Describe the enhancement you'd like**
Adding tags to the storefront filtering in addition to the type,vendor,price and variant options. I think this is required by majority of merchants

Regards
Muthu

**Comment by JamieBradders** [comment]
Agreed! I've often needed something like this for themes. Although I think this is a Shopify feature rather than something specific to Dawn? Can anyone recommend an alternative solution for tags? Is the Storefront API the best approach?

**Comment by sja87** [comment]
I also need a solution for tag or metafiled filtering. At the moment this is the only reason why we cannot switch to the new dawn theme.

**Comment by bredowmax** [comment]
Need it as well

**Comment by oksushi** [comment]
See further discussion here: https://github.com/Shopify/storefront-api-feedback/discussions/10

**Comment by MasjZam** [comment]
Although not tag support, the Online Store _does_ now support filtering by metafields as documented [here](https://help.shopify.com/en/manual/online-store/themes/os20/customize/filters#custom-filters). 

This allows you to include custom filters beyond what is already included in a product's variant options. This gives you a more structured solution with type and value validation that isn't possible with tags.

**Comment by mkromann** [comment]
> Although not tag support, the Online Store _does_ now support filtering by metafields as documented [here](https://help.shopify.com/en/manual/online-store/themes/os20/customize/filters#custom-filters).
> 
> This allows you to include custom filters beyond what is already included in a product's variant options. This gives you a more structured solution with type and value validation that isn't possible with tags.

Hi @MasjZam 

That is fantastic — a much better solution in my opinion. Unfortunately, there are some issues that make it unusable for many stores.

1. I tested it out the other day, and couldn't get a comma-separated list to work as an input. It is documented as a feature here: https://shopify.dev/themes/navigation-search/filtering#:~:text=Metafield,or%20canada%2Cusa.

It has also been reported here:  #381 

For many, this is a required feature. There are many cases where you need to include a product/variant in multiple groups. 

2. What is up with the 1000 products limit? This is another case where you either cannot use storefront filtering, OR you are too scared to commit, as you might outgrow the feature (this is the case for us). I understand that this is an API thing and not related to Dawn specifically but it has been reported here: #89 — but it has also been asked about numerous times here: https://github.com/Shopify/storefront-api-feedback/discussions/10. 

This limit is especially peculiar, as big collections are the ones that require filtering the most. Why is this limit in place? And is it something that you are looking to increase? Or is this how it is gonna for the foreseeable future?

——

I really appreciate these improvements and hope they can be further improved :)

**Comment by micahkala** [comment]
going to jump in here. i know this isn't theme related, but it may be backend related? but I am not able to get variant metafields to show up... I think there might be something wrong with either the documentation or the implementation... documentation says you can do filter.v.m.my_fields.fabric=leather, however, when I try it on my store, I am not able to get any results...
